### PR TITLE
Add GitHub Action to deploy site

### PR DESCRIPTION
**Pull request by msvamp**

### CODE REVIEW

**Comment by msvamp** [comment]
Oops, looks like ubuntu doesn't have `lftp` installed by default 😞

**Comment by msvamp** [comment]
Eh?
Command succeeded but nothing got uploaded?
(And entire job done in 10 seconds!)

**Comment by msvamp** [comment]
I guess I understood the problem now.
Apparently the current working directory for the workflow does not have files from the git repo. 😞

**Comment by msvamp** [comment]
Shit, I realised I didn't read the blank GitHub Actions template properly 🤦🏽

```YML
# Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
- uses: actions/checkout@v2
```

And I simply removed this line, silly me 🙃

**Comment by msvamp** [comment]
**SUCCESS!** 🏆
(I hope I didn't leak any sensetive info in the logs 😉)

File: .github/workflows/deploy-master.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  deploy-master-job:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install lftp
+      run: sudo apt install -y --no-install-recommends lftp >/dev/null
+
+    - name: Write the FTP reverse mirror command to a file
+      run: |
+        rm -rfv /dev/shm/somecommand
+        echo '#!/bin/bash'							 >/dev/shm/somecommand
```
Comment (msvamp) [inline .github/workflows/deploy-master.yml:25]: Um, I guess this line wasn't necessary 😶

File: .github/workflows/deploy-master.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        echo ' set ftp:list-options -a'				>>/dev/shm/somecommand
+        echo " set ftp:anon-user $SECRET2"			>>/dev/shm/somecommand
+        echo " set ftp:anon-pass $SECRET3"			>>/dev/shm/somecommand
+        echo ' mirror -cesRLv \'					>>/dev/shm/somecommand
+        echo '   -x^\.git/$ -x^\.github/$ \'		>>/dev/shm/somecommand
+        echo '   -X **/*.{MD,mD,Md,md} \'			>>/dev/shm/somecommand
+        echo "   . domains/$DDOMAIN/public_html"	>>/dev/shm/somecommand
+        echo ' quit'								>>/dev/shm/somecommand
+        echo '"'									>>/dev/shm/somecommand
+        chmod 0500 /dev/shm/somecommand
```
Comment (msvamp) [inline .github/workflows/deploy-master.yml:38]: Neither was this line required 🤦🏽

File: .github/workflows/deploy-master.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Fetch the last commit from repository
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+        clean: false
+
+    - name: Change to repo directory and run command from the file
+      run: |
+        cd "$GITHUB_WORKSPACE"
+        sh /dev/shm/somecommand
```
Comment (msvamp) [inline .github/workflows/deploy-master.yml:49]: Lol, silly me again 🙃
I set the interpreter to `bash` above (although not required) and ran the file with `sh` 😆

**Review by dev-kasibhatla — APPROVED** [review]
ManSamVam op

**Pull request merged**